File: web/sun-admin-web/src/types/department.ts
import type { AuditableEntity, EntityStatus } from './api';

export interface DepartmentItem extends AuditableEntity {
  parentId?: number | null;
  code: string;
  name: string;
  leader?: string;
  phone?: string;
  email?: string;
  sortOrder: number;
  status: EntityStatus;
  isBuiltIn?: boolean;
  children?: DepartmentItem[];
}

export interface DepartmentUpsertRequest {
  parentId?: number | null;
  code: string;
  name: string;
  leader?: string;
  phone?: string;
  email?: string;
  sortOrder: number;
  status?: EntityStatus;
}

File: src/SunAdmin.Domain/Entities/Department.cs
using SunAdmin.Domain.Common;
using SunAdmin.Domain.Enums;

namespace SunAdmin.Domain.Entities;

public sealed class Department : AuditableEntity
{
    public long? ParentId { get; set; }
    public string Code { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Leader { get; set; }
    public string? Phone { get; set; }
    public string? Email { get; set; }
    public int SortOrder { get; set; }
    public RecordStatus Status { get; set; } = RecordStatus.Enabled;
    public bool IsBuiltIn { get; set; }
}
